### PR TITLE
Remove Docker Desktop conflict for Rancher Desktop

### PR DESCRIPTION
**Pull request by sanyer**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

**Comment by neutric** [comment]
Could you please provide some context to your changes?

**Comment by sanyer** [comment]
@neutric yeah, sure.
Docker Desktop and Rancher Desktop are not mutually exclusive, although similar in function.

Situation is further improved by Rancher not installing supporting binaries (docker, kubectl, helm, etc) by default and allowing user to chose which ones to install.

**Comment by miccal** [comment]
Some background:
https://github.com/rancher-sandbox/rancher-desktop/issues/1081
https://github.com/rancher-sandbox/rancher-desktop/issues/1221
https://github.com/rancher-sandbox/rancher-desktop/issues/1300

**Comment by commitay** [comment]
This shouldn't be merged. 

The casks do conflict depending on which options are enabled in Rancher Desktop and as Homebrew doesn't have any knowledge about those options they need to be handled as if they have been enabled.

File: Casks/rancher.rb [hunk 1]
```diff
@@ -16,21 +16,13 @@
...
+    helm
+    kubernetes-cli
+  ]
 
   app "Rancher Desktop.app"
-  binary "#{appdir}/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/docker"
-  binary "#{appdir}/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/helm"
-  binary "#{appdir}/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/kim"
-  binary "#{appdir}/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/kubectl"
-  binary "#{appdir}/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/nerdctl"
```
Comment (commitay) [inline Casks/rancher.rb:33]: These binaries still need to be managed by Homebrew so they are removed if the cask is uninstalled, see https://github.com/rancher-sandbox/rancher-desktop/issues/272#issuecomment-931028377.

**Comment by miccal** [comment]
Thank you @sanyer but considering all of the above I have decided that this change should not be merged. If users want to install both this and Docker, I would suggest they install one of them manually and make their own choices to ensure that they do not conflict.

**Comment by sanyer** [comment]
@miccal interesting, thanks for the background.
I will check again how they both work for me.

**Comment by sanyer** [comment]
@commitay the issues you mentioned do exist, but they affect only the execution of these container management tools, not installation.

**Comment by sanyer** [comment]
Anyways, thank you both for the fast and useful feedback. I'll think about creating my own Tap for these both. Looks like a good candidate for trying out Taps :-)

**Comment by commitay** [comment]
Just mentioning this here in advance as I'm sure someone will open another PR for this ...

> https://docs.rancherdesktop.io/faq/
>
> Q: Can I have Docker Desktop installed alongside Rancher Desktop?
>
> A: Yes, but they cannot be run at the same time as both Rancher Desktop and Docker Desktop use the same Docker socket (/var/run/docker.sock). Before starting one, be sure to stop the other first.

Despite what their FAQ says it still conflicts as far as Homebrew is concerned as uninstalling/reinstalling/upgrading one cask may remove files that are used by the other cask.

The other option would be adding a `caveat` explaining all the above but people usually don't bother reading them anyways ...

**Comment by sanyer** [comment]
Here is my personal Tap for Docker Desktop and Rancher Desktop: https://github.com/sanyer/homebrew-containers-desktop

**Pull request closed without merging**